upi/vsphere/cluster_domain/main.tf

```diff
@@ -0,0 +1,36 @@
+locals {
+  lb_hostnames = ["api", "api-int", "*.apps"]
+}
+
+data "aws_route53_zone" "base" {
+  name = var.base_domain
+}
+
+resource "aws_route53_zone" "cluster" {
+  name          = var.cluster_domain
+  force_destroy = true
+
+  tags = {
+    "Name"     = var.cluster_domain
+    "Platform" = "vSphere"
+  }
+}
+
+resource "aws_route53_record" "name_server" {
+  name    = var.cluster_domain
+  type    = "NS"
+  ttl     = "300"
+  zone_id = data.aws_route53_zone.base.zone_id
+  records = aws_route53_zone.cluster.name_servers
+}
```

Comment:
nit: in the future, we should drop these resources, as we can use the `data.aws_route53_zone.base.zone_id` to create all our records. we don't need a separate route53 zone.

Reply:
and when we do, we should drop this module all together